File: 110. Balanced Binary Tree/AC_Answer_1.py
# O(n^2) time use bfs

# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def depth(self, root):
        if root==None: return 0
        return max(depth(root.left),depth(root.right))+1
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root==None: return True
        left = depth(root.left)
        right = depth(root.right)
        return abs(left-right)==1 and self.isBalanced(root.left) and self.isBalanced(root.right)
        

File: 653. Two Sum IV- input is BST/AC_Answer_1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean findTarget(TreeNode root, int k) {
        List<Integer> list = new ArrayList<>();
        findValues(list, root);
        for(Integer i:list){
            if(list.contains(k-i) && k-i!=i) return true;
        }
        return false;
    }
    
    public void findValues(List<Integer> list, TreeNode root){
        if(root==null) return;
        list.add(root.val);
        findValues(list, root.left);
        findValues(list, root.right);
    }
}
